src/super_gradients/training/utils/detection_utils.py

```diff
@@ -768,12 +759,8 @@ def __call__(self, data) -> Tuple[torch.Tensor, torch.Tensor, Dict[str, torch.Te
         if self.random_resize_sizes is not None:
             data = self.random_resize(data)
 
-        batch = default_collate(data)
-        ims, targets, crowd_targets = batch
-        if ims.shape[3] == 3:
-            ims = torch.moveaxis(ims, -1, 1).float()
-
-        return ims, self._format_targets(targets), {"crowd_targets": self._format_targets(crowd_targets)}
+        images_batch, labels_batch, crowd_labels_batch = list(zip(*data))
+        return self._format_images(images_batch), self._format_targets(labels_batch), {"crowd_targets": self._format_targets(crowd_labels_batch)}
```

Comment:
Same idea as above, but reversed: It would be great to raise an error if `len(batch) != 3`
I know we did not in the past, but this was a mistake, the code would just break without any explanation of why it doesnt work.

Reply:
The code will break with "too many/not enough values to unpack", do you mean a more verbose and explanatory error?

Reply:
Yeah, not for me but for others who would be lost with this error.

Reply:
See the answer above